### PR TITLE
ddl: fix wrong error message when renaming non-exist table

### PR DESCRIPTION
**Pull request by c4pt0r**
Fix for issue #29893

### CODE REVIEW

**Comment by hawkingrei** [comment]
LGTM

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -5161,7 +5161,7 @@ func extractTblInfos(is infoschema.InfoSchema, oldIdent, newIdent ast.Ident, isA
 		if tableExists(is, newIdent, tables) {
 			return nil, 0, infoschema.ErrTableExists.GenWithStackByArgs(newIdent)
 		}
-		return nil, 0, errFileNotFound.GenWithStackByArgs(oldIdent.Schema, oldIdent.Name)
+		return nil, 0, infoschema.ErrTableNotExists.GenWithStackByArgs(oldIdent.Schema, oldIdent.Name)
```
Comment (xiongjiwei) [inline ddl/ddl_api.go:5164]: there is another `errFileNotFound` usage at L5155, can you please remove it?

**Review by xiongjiwei — APPROVED** [review]

**Review by lcwangchao — APPROVED** [review]

**Comment by zhangyangyu** [comment]
Doesn't seems like unstable tests. There seems several tests relying on the specific error. And if change https://github.com/pingcap/tidb/blob/177be48e9483db48f5da7f5f2a8ab67368c923cf/ddl/ddl_api.go#L5164, then how about https://github.com/pingcap/tidb/blob/177be48e9483db48f5da7f5f2a8ab67368c923cf/ddl/ddl_api.go#L5155 ?

**Comment by bb7133** [comment]
https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/46508/pipeline

Please update the test case:

```
FAIL: ddl_test.go:1567: testRecoverTable.TestRenameMultiTables
[2021-11-19T10:06:58.105Z] 
[2021-11-19T10:06:58.105Z] ddl_test.go:1614:
[2021-11-19T10:06:58.105Z]     tk.MustGetErrCode("rename table rename1.t1 to rename1.t2, rename1.t1 to rename3.t4", errno.ErrFileNotFound)
[2021-11-19T10:06:58.105Z] /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/util/testkit/testkit.go:407:
[2021-11-19T10:06:58.105Z]     tk.c.Assert(int(sqlErr.Code), check.Equals, errCode, check.Commentf("Assertion failed, origin err:\n  %v", sqlErr))
[2021-11-19T10:06:58.105Z] ... obtained int = 1146
[2021-11-19T10:06:58.105Z] ... expected int = 1017
[2021-11-19T10:06:58.105Z] ... Assertion failed, origin err:
[2021-11-19T10:06:58.105Z]   ERROR 1146 (42S02): Table 'rename1.t1' doesn't exist
```

**Comment by c4pt0r** [comment]
@bb7133 Ok, will do.

**Comment by feitian124** [comment]
thanks to tests..

**Comment by hawkingrei** [comment]
It is fixed by #32170

**Pull request closed without merging**